### PR TITLE
Add support for rewindable transforms

### PR DESCRIPTION
**Pull request by smoogipoo**
Property added:
* `RemoveCompletedTransforms`: Basically, any time either: a) a transform was applied while another of the same type was active, or b) a transform was "complete", the transforms would be removed from the Transformable. This blocks that.
  * Publicly-overridable: Containers are able to override this property to set a custom value.
  * Internally-settable: CompositeDrawable passes down the value recursively when children are added.

### CODE REVIEW

File: osu.Framework/Graphics/Transforms/Transformable.cs [hunk 1]
```diff
@@ -69,6 +74,27 @@ private void updateTransforms()
             if (transformsLazy == null)
                 return;
 
+            if (!RemoveCompletedTransforms)
+            {
+                // Under the case that completed transforms are not removed, reversing the clock is permitted.
+                // We need to first look back through all the transforms and apply the start values of the ones that were previously
+                // applied, but now exist in the future relative to the current time.
```
Comment (Tom94) [inline osu.Framework/Graphics/Transforms/Transformable.cs]: It's unfortunate that we can not know the _actual_ value a drawable had at the beginning of those already applied transforms. Perhaps we can add an additional Field in each `Transform` which stores the current value at the point of first application? (I.e. if `Applied == false`)

File: osu.Framework/Graphics/Transforms/Transformable.cs [hunk 2]
```diff
@@ -291,6 +324,7 @@ public void AddTransform(Transform transform)
             if (Clock == null)
             {
                 transform.Apply(transform.EndTime);
+                transform.Applied = true;
```
Comment (Tom94) [inline osu.Framework/Graphics/Transforms/Transformable.cs]: I think I would prefer if `Apply` automatically set `Applied` to true.

**Comment by Tom94** [comment]
In general, I am happy this turned out to require only such a small amount of code. Means the underlying mechanism for storing and managing `Transform`s is somewhat solid. :)

File: osu.Framework/Graphics/Transforms/Transformable.cs (hunk 1 above)
Comment (smoogipoo) [reply osu.Framework/Graphics/Transforms/Transformable.cs]: As discussed in discord - this is not an issue as StartValue is already initialised to the value right before it is first applied ( https://github.com/smoogipoo/osu-framework/blob/master/osu.Framework/Graphics/Transforms/Transformable.cs#L79-L82 )

**Pull request merged**